app/controllers/api/v1/users_controller.rb

```diff
@@ -29,6 +29,51 @@ def update
         end
       end
 
+      def exist
+        userData = params[:user]
+        Rails.logger.info "************ exist is = #{userData[:email]}"
+        user = User.where(email: userData[:email]).first
+        Rails.logger.info "************ user is = #{user}"
+        @redirect_path = '/profile' #change this to actually redirect to the social login function
+        unless user
+             @redirect_path = '/additional_info'
+        end
+        Rails.logger.info "!!!!!!!!!!!! path is #{@redirect_path}"
+        render json: {
+          redirect_to: @redirect_path
+        }
+      end
+
+      def social
+        Rails.logger.info "************ got to create = #{params.inspect}"
+          path = ''
+          arr = User.from_social(params[:user])
```

Comment:
Can we abstract lines 50 - 52 out into a private method?  For example, `set_social_user`, or something like that?

Reply:
done.